### PR TITLE
Replace Units with UnitsNet

### PR DESCRIPTION
**Pull request by krwq**
**Note: This PR is against feature-units branch**

This is a prototype/experiment work which is part of https://github.com/dotnet/iot/issues/869

- Removes Units
- Adds package reference to UnitsNet
- Adjust code to make it build

I'm still voting for yes on this library

I've run our Units tests before removing them against this library. There are some differences with formatting and treating NaN and Infinities.

cc: @joperezr @pgrawehr @MarkCiliaVincenti @JTrotta @Ellerbach 

<details>

<summary>Test run output</summary>

```
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:00.82]     Units.Tests.PressureTests.ConversionFactorTests [FAIL]
[xUnit.net 00:00:00.82]     Iot.Units.Tests.TemperatureTests.FromCelsiusPositiveInfinity [FAIL]
[xUnit.net 00:00:00.82]     Iot.Units.Tests.TemperatureTests.FromKelvinNaN [FAIL]
  X Units.Tests.PressureTests.ConversionFactorTests [6ms]
  Error Message:
   Assert.Equal() Failure
Expected: 102423.94009 (rounded from 102423.940094101)
Actual:   102424 (rounded from 102424)
  Stack Trace:
     at Units.Tests.PressureTests.ConversionFactorTests() in C:\src\iot\src\devices\Units\tests\PressureTests.cs:line 21
  X Iot.Units.Tests.TemperatureTests.FromCelsiusPositiveInfinity [1ms]
  Error Message:
   System.ArgumentException : PositiveInfinity or NegativeInfinity is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromCelsiusPositiveInfinity() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 78
  X Iot.Units.Tests.TemperatureTests.FromKelvinNaN [1ms]
  Error Message:
   System.ArgumentException : NaN is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromKelvinNaN() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 69
[xUnit.net 00:00:00.82]     Iot.Units.Tests.TemperatureTests.FromFahrenheitNegativeInfinity [FAIL]
[xUnit.net 00:00:00.82]     Iot.Units.Tests.TemperatureTests.FromKelvinNegativeInfinity [FAIL]
[xUnit.net 00:00:00.83]     Iot.Units.Tests.TemperatureTests.FromFahrenheitNaN [FAIL]
[xUnit.net 00:00:00.83]     Iot.Units.Tests.TemperatureTests.FromCelsiusNegativeInfinity [FAIL]
[xUnit.net 00:00:00.83]     Iot.Units.Tests.TemperatureTests.ToStringWithFormatArgs [FAIL]
[xUnit.net 00:00:00.83]     Iot.Units.Tests.TemperatureTests.FromCelsiusNaN [FAIL]
[xUnit.net 00:00:00.83]     Iot.Units.Tests.TemperatureTests.ToStringWithFormatArgsAndCulture [FAIL]
[xUnit.net 00:00:00.83]     Iot.Units.Tests.TemperatureTests.FromKelvinPositiveInfinity [FAIL]
  X Iot.Units.Tests.TemperatureTests.FromFahrenheitNegativeInfinity [1ms]
  Error Message:
   System.ArgumentException : PositiveInfinity or NegativeInfinity is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromFahrenheitNegativeInfinity() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 114
  X Iot.Units.Tests.TemperatureTests.FromKelvinNegativeInfinity [1ms]
  Error Message:
   System.ArgumentException : PositiveInfinity or NegativeInfinity is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromKelvinNegativeInfinity() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 123
  X Iot.Units.Tests.TemperatureTests.FromFahrenheitNaN [1ms]
  Error Message:
   System.ArgumentException : NaN is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromFahrenheitNaN() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 60
  X Iot.Units.Tests.TemperatureTests.FromCelsiusNegativeInfinity [1ms]
  Error Message:
   System.ArgumentException : PositiveInfinity or NegativeInfinity is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromCelsiusNegativeInfinity() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 105
  X Iot.Units.Tests.TemperatureTests.ToStringWithFormatArgs [4ms]
  Error Message:
   System.FormatException : The C format string is not supported.
  Stack Trace:
     at UnitsNet.QuantityFormatter.Format[TUnitType](IQuantity`1 quantity, String format, IFormatProvider formatProvider) in C:\projects\unitsnet\UnitsNet\QuantityFormatter.cs:line 76
   at Iot.Units.Tests.TemperatureTests.ToStringWithFormatArgs() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 153
  X Iot.Units.Tests.TemperatureTests.FromCelsiusNaN [1ms]
  Error Message:
   System.ArgumentException : NaN is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromCelsiusNaN() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 51
  X Iot.Units.Tests.TemperatureTests.ToStringWithFormatArgsAndCulture [1ms]
  Error Message:
   System.FormatException : The K2 format string is not supported.
  Stack Trace:
     at UnitsNet.QuantityFormatter.Format[TUnitType](IQuantity`1 quantity, String format, IFormatProvider formatProvider) in C:\projects\unitsnet\UnitsNet\QuantityFormatter.cs:line 76
   at Iot.Units.Tests.TemperatureTests.ToStringWithFormatArgsAndCulture() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 170
  X Iot.Units.Tests.TemperatureTests.FromKelvinPositiveInfinity [1ms]
  Error Message:
   System.ArgumentException : PositiveInfinity or NegativeInfinity is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromKelvinPositiveInfinity() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 96
[xUnit.net 00:00:00.83]     Iot.Units.Tests.TemperatureTests.FromFahrenheitPositiveInfinity [FAIL]
[xUnit.net 00:00:00.90]     Iot.Units.Tests.TemperatureTests.ToStringDefault [FAIL]
[xUnit.net 00:00:00.90]     Units.Tests.PressureTests.ToStringDefault [FAIL]
  X Iot.Units.Tests.TemperatureTests.FromFahrenheitPositiveInfinity [1ms]
  Error Message:
   System.ArgumentException : PositiveInfinity or NegativeInfinity is not a valid number.
Parameter name: val
  Stack Trace:
     at UnitsNet.InternalHelpers.Guard.EnsureValidNumber(Double value, String paramName) in C:\projects\unitsnet\UnitsNet\InternalHelpers\Guard.cs:line 25
   at UnitsNet.QuantityValue..ctor(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 39
   at UnitsNet.QuantityValue.op_Implicit(Double val) in C:\projects\unitsnet\UnitsNet\QuantityValue.cs:line 66
   at Iot.Units.Tests.TemperatureTests.FromFahrenheitPositiveInfinity() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 87
  X Iot.Units.Tests.TemperatureTests.ToStringDefault [105ms]
  Error Message:
   Assert.Equal() Failure
               (pos 4)
Expected: 20.2°C
Actual:   20.21 °C
               (pos 4)
  Stack Trace:
     at Iot.Units.Tests.TemperatureTests.ToStringDefault() in C:\src\iot\src\devices\Units\tests\TemperatureTests.cs:line 136
  X Units.Tests.PressureTests.ToStringDefault [126ms]
  Error Message:
   Assert.Equal() Failure
            (pos 1)
Expected: 1024.24hPa
Actual:   1'024.24 hPa
            (pos 1)
  Stack Trace:
     at Units.Tests.PressureTests.ToStringDefault() in C:\src\iot\src\devices\Units\tests\PressureTests.cs:line 50
```

</details>

### CODE REVIEW

**Comment by krwq** [comment]
Note, CI is not enabled here although this is building and passing Unit tests on my Windows box.

File: src/devices/Common/Iot/Device/Common/WeatherHelper.cs [hunk 1]
```diff
@@ -12,6 +12,11 @@ namespace Iot.Device.Common
     /// </summary>
     public static class WeatherHelper
     {
+        /// <summary>
+        /// The mean sea-level pressure (MSLP) is the average atmospheric pressure at mean sea level
+        /// </summary>
+        public static readonly Pressure MeanSeaLevel = Pressure.FromPascals(101325);
```
Comment (krwq) [inline src/devices/Common/Iot/Device/Common/WeatherHelper.cs:18]: I've moved this API here. Couldn't find constants in UnitsNet

**Comment by pgrawehr** [comment]
The formatting differences are negligible IMHO (just different number of decimals by default). 
The different behavior when converting NAN values will require some extra considerations. UnitsNet doesn't allow storing invalid numbers in their units [Here is](https://github.com/angularsen/UnitsNet/issues/695) some background on it. Unfortunatelly, that thread ends with the discussion about an alternate approach, not with a solution/suggestion for the OPs problem. We'll probably have to use Nullable return values where we previously returned NaN. Or restart that thread over there.

**Comment by krwq** [comment]
@pgrawehr I don't think we've ever recommended returning NaN as replacement for "no value" - we usually prefer `TryGet*` pattern for that

**Comment by pgrawehr** [comment]
I've at least seen several examples where TryGet() was used, but the return value ignored and NaN expected for invalid results. But I agree that one should not depend on the value returned by a TryGet() when the return value is false.

**Comment by krwq** [comment]
@pgrawehr since the units are structs in UnitsNet you can just use `default` or `default(Pressure)` and currently looking at the implementation this will give you invalid value:
https://github.com/angularsen/UnitsNet/blob/master/UnitsNet/GeneratedCode/Quantities/Pressure.g.cs#L45 (Unit will be null internally and cause all kinds of errors on usage)

**Comment by krwq** [comment]
I'm merging this since this is not against master.

**Comment by joperezr** [comment]
> Note, CI is not enabled here although this is building and passing Unit tests on my Windows box.

I've sent out #1053 to add coverage

**Pull request merged**